dftimewolf/lib/processors/gcp_cloud_resource_tree_helper.py

```diff
@@ -266,54 +273,116 @@ def GenerateChildrenTree(self, level: int) -> List[Dict[str, 'Resource']]:
 
   def __str__(self) -> str:
     """Returns a string representation of the resource tree."""
-    output = '\n'
+    output = io.StringIO('')
+    # output = '\n'
     dashes = '-' * 200
-    # pylint: disable=line-too-long
-    output_format = '{:<20s}|{:<18s}|{:<20s}|{:<25s}|{:<18s}|{:<20s}|{:<25s}|{:<18s}|{:<100s}\n'
-
     # Draw table header
-    output = output + dashes + '\n'
-    output = output + output_format.format(
-        'Resource ID', 'Resource Type', 'Creation TimeStamp', 'Created By',
-        'Creator IP Addr', 'Deletion Timestamp', 'Deleted By',
-        'Deleter IP Addr', 'Tree')
-    output = output + dashes + '\n'
-
+    output.write(f'\n{dashes}\n')
+    output.write(f"""
+                 {"Resource ID":<20s}|
+                 {"Resource Type":<18s}|
+                 {"Creation TimeStamp":<20s}|
+                 {"Created By":<25s}|
+                 {"Creator IP Addr":<18s}|
+                 {"Deletion Timestamp":<20s}|
+                 {"Deleted By":<25s}|
+                 {"Deleter IP Addr":<18s}|
+                 {"Tree":<100s}\n
+                 """)
+    output.write(f'{dashes}\n')
+
+    # Generate resource tree
     result = self.GenerateTree()
-    for i in result:
-      resource: Optional[Resource] = i.get('resource_object')
+
+    # Draw table rows
+    for entry in result:
+      resource: Optional[Resource] = entry.get('resource_object')
       if resource:
-        output = output + \
-            (output_format.format( resource.id if ('-' not in resource.id) else "" , resource.type, resource.creation_timestamp.astimezone(timezone.utc).strftime("%Y-%m-%d %H:%M:%S") if resource.creation_timestamp else "",
-            resource.created_by, resource.creator_ip_address,  resource.deletion_timestamp.astimezone(timezone.utc).strftime("%Y-%m-%d %H:%M:%S") if resource.deletion_timestamp else "", resource.deleted_by, resource.deleter_ip_address,  i.get('graph')))
-    output = output + dashes + '\n'
 
-    return output
+        output.write(f"""
```

Comment:
as i understand multiline strings include newlines/spaces in the string so where you have introduced this the output isn't the same as what has been replaced.  checking that this is still WAI?

Reply:
Fixed